src/AddressRegistry.Api.Legacy/AddressMatch/Matching/Sanitizer.cs

```diff
@@ -5,376 +5,581 @@ namespace AddressRegistry.Api.Legacy.AddressMatch.Matching
-                    if (FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam)) == "0000"
-                        && FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)) != null
-                        && Int32.TryParse(FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)), out hnr)
+                    if (possibleIndexInStreet == "0000"
+                        && possibleNumberInStreet != null
+                        && int.TryParse(possibleNumberInStreet, out hnr)
                         && hnr != 0)
                     {
-                        houseNumber0 = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
+                        formattedHouseNumber = possibleNumberInStreet;
                     }
                 }
-                //huisnummer met niet-numeriek bisnummer als prefix in straatnaam
+
+                // huisnummer met niet-numeriek bisnummer als prefix in straatnaam
+                // TODO: [0-9]+ ?
                 if (new Regex("[0-9][a-zA-Z] ").IsMatch(streetName) && !new Regex("[0-9]e ").IsMatch(streetName))
                 {
-                    if (FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)) != null
-                        && Int32.TryParse(FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)), out hnr)
+                    if (possibleNumberInStreet != null
+                        && int.TryParse(possibleNumberInStreet, out hnr)
                         && hnr != 0
+                        // TODO: [0-9]+ ?
                         && new Regex("[0-9][a-zA-Z]$").IsMatch(requestStraatnaamWithoutStraatnaam))
                     {
-                        houseNumber0 = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
-                        rrindex0 = FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam)).ToUpper();
+                        formattedHouseNumber = possibleNumberInStreet;
+                        formattedRrIndex = possibleIndexInStreet.ToUpper();
                     }
                 }
-                //huisnummer als suffix in straatnaam
+
+                // huisnummer als suffix in straatnaam
                 if (new Regex(" [0-9]+$").IsMatch(streetName))
                 {
-                    houseNumber0 = FormatRrHouseNumber(new Regex(" [0-9]+$").Match(streetName).Value);
-                    //TODO Stif: ask business, why only here warning?
+                    formattedHouseNumber = FormatRrHouseNumber(new Regex(" [0-9]+$").Match(streetName).Value);
                     Warnings.AddWarning("2", "'Huisnummer' in 'Straatnaam' gevonden.");
                 }
-                //huisnummer met niet-numeriek bisnummer als suffix in straatnaam
+
+                // huisnummer met niet-numeriek bisnummer als suffix in straatnaam
+                // TODO: We dont care about excluding [0-9]e anymore?
                 if (new Regex(" [0-9]+[a-zA-Z]$").IsMatch(streetName))
```

Comment:
We dont care about excluding `[0-9]e` anymore here?

We did before:
`if (new Regex("[0-9][a-zA-Z] ").IsMatch(streetName) && !new Regex("[0-9]e ").IsMatch(streetName))`